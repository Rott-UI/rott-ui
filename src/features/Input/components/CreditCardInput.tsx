// React Imports
import {FC} from 'react'

// React Native Imports
import {StyleSheet} from 'react-native'

// Component Imports
import {InputProps} from '../models'
import {InputStyles} from '../styles'

// Package Imports
import {MaskedTextInput} from 'react-native-mask-text'

export const CreditCardInput: FC<Omit<InputProps, 'label' | 'name'>> = ({
  fontSize,
  onChangeText,
  theme,
  disabled,
  size,
  ...props
}) => {
  const handleOnChangeText = (text: string) => {
    const replacedText = text.replace(/[^0-9]/g, '')
    onChangeText!(replacedText)
  }

  return (
    <MaskedTextInput
      editable={!disabled}
      mask='9999-9999-9999-9999'
      placeholder='**** **** **** ****'
      keyboardType='number-pad'
      maxLength={19}
      onChangeText={(text, rawText) => handleOnChangeText(rawText)}
      style={StyleSheet.flatten([InputStyles({fontSize, theme, size}).defaultTextInputStyle])}
      {...props}
    />
  )
}
