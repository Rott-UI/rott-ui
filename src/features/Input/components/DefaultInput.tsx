// React Imports
import {FC} from 'react'

// React Native Imports
import {StyleSheet, TextInput} from 'react-native'

// Component Imports
import {InputProps} from '../models'
import {InputStyles} from '../styles'

export const DefaultInput: FC<Omit<InputProps, 'name'>> = ({
  label,
  placeholder,
  fontSize,
  theme,
  size,
  ...props
}) => {
  return (
    <TextInput
      keyboardType='default'
      autoCapitalize='none'
      placeholder={placeholder ?? (typeof label === 'string' ? label : undefined)}
      style={StyleSheet.flatten([InputStyles({fontSize, theme, size}).defaultTextInputStyle])}
      {...props}
    />
  )
}
