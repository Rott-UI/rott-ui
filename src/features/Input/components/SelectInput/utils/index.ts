export * from './listHeightNormalizer'
export * from './modalHeightPercentageNormalizer'
