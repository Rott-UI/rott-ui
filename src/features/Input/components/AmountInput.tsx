// React Imports
import {FC, useCallback, useEffect, useRef, useState} from 'react'

// React Native Imports
import {StyleSheet, TextInput} from 'react-native'

// Component Imports
import {InputProps} from '../models'
import {InputStyles} from '../styles'
import {Item} from '../../Item'
import {Label} from '../../Label'
import {Icon} from '../../Icon'

// Util and Lib Imports
import {themeConfig} from '../../../providers'

// Package Imports
import {AmountInputStyles} from '../styles/AmountInput.style'
import {InputStyleNormalizer} from '../utils/inputNormalizer'

export const AmountInput: FC<Omit<InputProps, 'placeholder' | 'label' | 'name'>> = ({
  fontSize,
  onChangeText,
  value,
  theme,
  disabled,
  size,
  ...props
}) => {
  const amountRef = useRef<TextInput>(null)
  const [amount, setAmount] = useState('')
  const [currency, setCurrency] = useState('')

  const resetInternalStates = useCallback(() => {
    setAmount('')
    setCurrency('')
    onChangeText!('0.00')
  }, [])

  const replaceTextWithNumberOrEmpty = (text: string) =>
    text.replace(/[^0-9]/g, '').replace(/^0+/, '')

  const amountNormalizer = (text: string) => {
    if (!text) return resetInternalStates()

    const replacedAmount = replaceTextWithNumberOrEmpty(text)
    const reversedText = replacedAmount.split('').reverse().join('') ?? ''
    const dottedText = reversedText?.match(/.{1,3}/g)?.join('.')
    const normalizedText = dottedText?.split('').reverse().join('')

    setAmount(normalizedText ?? '')
  }

  const handleTextChange = (inputTextAmount: string, inputTextCurrency: string) => {
    if (!inputTextAmount && !inputTextCurrency) return

    let amountFormat = replaceTextWithNumberOrEmpty(inputTextAmount)
    let currencyFormat = replaceTextWithNumberOrEmpty(inputTextCurrency)

    amountFormat = amountFormat.length === 0 ? '0' : amountFormat

    currencyFormat = currencyFormat.length === 0 ? '00' : currencyFormat
    currencyFormat = currencyFormat.length === 1 ? `${currencyFormat}0` : currencyFormat

    onChangeText!(`${amountFormat}.${currencyFormat}`)
  }

  const placeholderColorNormalizer =
    theme === 'dark'
      ? themeConfig.colors['white']
      : (!amount && !currency) ||
        ((amount === '0' || amount === '') && (!currency || currency === '' || currency === '00'))
      ? themeConfig.colors['grey-200']
      : themeConfig.colors['grey-900']

  useEffect(() => {
    const initialAmount = value?.split(',')[0]
    const initialCurrency = value?.split(',')[1]

    setAmount(initialAmount ?? '')
    setCurrency(initialCurrency ?? '')
  }, [])

  useEffect(() => {
    if (value === '' || !value || value === '0') resetInternalStates()
  }, [value])

  useEffect(() => {
    handleTextChange(amount, currency)
  }, [amount, currency])

  return (
    <Item relative onTouchStart={() => amountRef.current?.focus()}>
      <Item row {...props}>
        <TextInput
          ref={amountRef}
          nativeID='amount-native-id'
          testID='amount-test-id'
          editable={!disabled}
          placeholder='0'
          maxLength={11}
          value={amount}
          keyboardType='number-pad'
          onChangeText={(text) => amountNormalizer(text)}
          style={StyleSheet.flatten([
            InputStyles({fontSize, theme, size}).defaultTextInputStyle,
            AmountInputStyles().amountInputStyle,
            {
              color: placeholderColorNormalizer,
            },
          ])}
          placeholderTextColor={placeholderColorNormalizer}
        />

        <Item
          justifyContentFlexEnd
          paddingBottom={InputStyleNormalizer({size}).bottomElementPadding}>
          <Label
            fontSize={
              fontSize ??
              InputStyleNormalizer({
                size,
              }).placeholderSize
            }
            fontWeight='bold'
            color={placeholderColorNormalizer as string}>
            ,
          </Label>
        </Item>

        <TextInput
          testID='currency-test-id'
          maxLength={2}
          placeholder='00'
          value={currency}
          keyboardType='number-pad'
          onChangeText={(text) => setCurrency(replaceTextWithNumberOrEmpty(text).substring(0, 2))}
          onFocus={() => (currency === '00' ? setCurrency('') : undefined)}
          onBlur={() => (currency.isEmpty() || currency === '0' ? setCurrency('00') : undefined)}
          style={StyleSheet.flatten([
            InputStyles({fontSize, theme, size}).defaultTextInputStyle,
            AmountInputStyles().amountInputStyle,
            {
              color: placeholderColorNormalizer,
            },
          ])}
          placeholderTextColor={placeholderColorNormalizer}
        />
      </Item>

      <Item absolute right={0} bottom={InputStyleNormalizer({size}).icon.paddingBottom}>
        <Icon
          name='TL'
          width={InputStyleNormalizer({size}).icon.width}
          height={InputStyleNormalizer({size}).icon.height}
          color={themeConfig.colors['grey-200']}
          mode='fill'
          noStroke
        />
      </Item>
    </Item>
  )
}
