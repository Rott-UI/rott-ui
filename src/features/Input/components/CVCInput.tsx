// React Imports
import {FC} from 'react'

// React Native Imports
import {Alert, StyleSheet, TextInput} from 'react-native'

// Component Imports
import {InputProps} from '../models'
import {InputStyles, CVCInputStyles} from '../styles'
import {Item} from '../../Item'
import {Icon} from '../../Icon'
import {Pressable} from '../../Pressable'

// Util and Lib Imports
import {formatMessage} from '../../../utils'

export const CVCInput: FC<Omit<InputProps, 'label' | 'placeholder' | 'name'>> = ({
  fontSize,
  onChangeText,
  theme,
  size,
  ...props
}) => {
  const handleTextChange = (inputText: string) => {
    let text = inputText.replace(/[^0-9]/g, '')
    text = text.length > 3 ? text.substring(0, 3) : text
    onChangeText!(text)
  }

  return (
    <Item row>
      <TextInput
        editable={!props.disabled}
        placeholder='***'
        style={StyleSheet.flatten([InputStyles({fontSize, theme, size}).defaultTextInputStyle])}
        keyboardType='number-pad'
        maxLength={3}
        onChangeText={handleTextChange}
        {...props}
      />

      {/* TODO: CVC Info butonuna kart arkayüzü eklenmeli */}
      <Pressable
        testID='info-icon-test-id'
        style={CVCInputStyles().infoIcon}
        justifyContentCenter
        alignItemsCenter
        onPress={() =>
          Alert.alert(formatMessage('CVC.INFO.TITLE'), formatMessage('CVC.INFO.DESCRIPTION'), [
            {
              text: formatMessage('COMMON.OK'),
              style: 'cancel',
            },
          ])
        }>
        <Icon name='CHECK_CIRCLE' height={24} width={24} variant='grey-900' />
      </Pressable>
    </Item>
  )
}
