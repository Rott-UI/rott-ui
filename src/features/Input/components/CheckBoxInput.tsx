// React Imports
import {FC, ReactNode} from 'react'

// React Native Imports

// Component Imports
import {InputProps} from '../models'
import {Label} from '../../Label'
import {Item} from '../../Item'
import {Pressable} from '../../Pressable'

// Util and Lib Imports
import {InputStyles} from '../styles'
import {InputStyleNormalizer} from '../utils/inputNormalizer'
import {themeConfig} from '../../../providers'

interface CheckBoxProps extends Omit<InputProps, 'placeholder' | 'name' | 'label'> {
  description?: ReactNode
  checked?: boolean
  onCheckChange?: (checked: boolean) => void
}

export const CheckBoxInput: FC<CheckBoxProps> = ({
  description,
  checked,
  fontSize,
  theme,
  size,
  disabled,
  onCheckChange,
}) => {
  return (
    <Pressable
      flex={0}
      testID='checkbox-input-test-id'
      onPress={() => !disabled && onCheckChange!(!checked)}>
      <Item
        row
        alignItemsCenter
        style={InputStyles({theme, size}).defaultTextInputStyle}
        testID='checkbox-container-test-id'>
        <Item
          borderRadius={8}
          borderColor={checked ? themeConfig.colors['primary'] : themeConfig.colors['grey-200']}
          borderWidth={2}
          backgroundColor='white'
          width={24}
          height={24}
          alignItemsCenter
          justifyContentCenter>
          {checked && (
            <Item
              testID='checkbox-checked-test-id'
              borderWidth={2}
              borderColor={themeConfig.colors['neutral-blue-alpha']}
              borderRadius={4}
              backgroundColor={themeConfig.colors['primary']}
              width={15}
              height={15}
            />
          )}
        </Item>

        {description && typeof description === 'string' && (
          <Label
            testID='checkbox-default-label-test-id'
            fontSize={fontSize ?? InputStyleNormalizer({size}).placeholderSize}
            fontWeight='bold'
            marginLeft={12}
            variant={theme === 'dark' ? 'white' : 'grey-900'}>
            {description}
          </Label>
        )}

        {description && typeof description !== 'string' && <>{description}</>}
      </Item>
    </Pressable>
  )
}
