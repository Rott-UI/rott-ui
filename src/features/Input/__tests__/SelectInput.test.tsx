// Component Imports
import {SelectInput} from '../components'

// Util and Lib Imports
import {act, fireEvent, render, waitFor} from '../../../utils'

describe('Select Input -> Custom Input', () => {
  const testId = {
    select: {
      selectTestId: 'select-test-id',
      selectSelectionTestId: 'select-input-selection-test-id',
      modalTestId: 'select-modal-test-id',
      listTestId: 'select-list-test-id',
      selectedTestId: 'select-input-selected-item-test-id',
      selectedDescriptionTestId: 'selected-description-test-id',
    },
    search: {
      searchInputTestId: 'select-search-input-test-id',
      searchClearTestId: 'select-search-clear-test-id',
    },
  }

  const defaultLabel = 'Label Text'

  const mockData = [
    {label: 'Apple', value: 'apple', description: 'Apple Desc'},
    {label: 'Samsung', value: 'samsung', description: 'Samsung Desc'},
    {label: 'Huawei', value: 'huawei', description: 'Huawei Desc'},
    {label: 'Oppo', value: 'oppo', description: 'Oppo Desc'},
    {label: 'Realme', value: 'realme', description: 'Realme Desc'},
    {label: 'LG', value: 'lg', description: 'LG Desc'},
    {label: 'Vestel', value: 'vestel', description: 'Vestel Desc'},
    {label: 'Motorola', value: 'motorola', description: 'Motorola Desc'},
    {label: 'Panasonic', value: 'panasonic', description: 'Panasonic Desc'},
    {label: 'Ericson', value: 'ericson', description: 'Ericson Desc'},
    {label: 'HTC', value: 'htc', description: 'HTC Desc'},
    {label: 'ASUS', value: 'asus', description: 'ASUS Desc'},
    {label: 'Lenovo', value: 'lenovo', description: 'Lenovo Desc'},
  ]

  it('ilk render anında snapshot ile eşleşmeli', () => {
    const {
      select: {selectTestId},
    } = testId
    const rendered = render(
      <SelectInput testID={selectTestId} list={mockData} label={defaultLabel} />
    )

    expect(rendered).toMatchSnapshot()
  })

  it('select inputa tıklandığında input modal olarak açılmalı', () => {
    const {
      select: {selectTestId, selectSelectionTestId, modalTestId},
    } = testId
    const {getByTestId} = render(
      <SelectInput testID={selectTestId} list={mockData} label={defaultLabel} />
    )

    // Select Modal Acilir
    const selectInputValueContainer = getByTestId(selectSelectionTestId)
    act(() => {
      fireEvent.press(selectInputValueContainer)
    })

    const selectInputModal = getByTestId(modalTestId)
    expect(selectInputModal).toBeVisible()
  })

  it('select input secildigi zaman listenin ekranda gorunur olmali', () => {
    const {
      select: {selectTestId, selectSelectionTestId, listTestId},
    } = testId
    const {getByTestId} = render(
      <SelectInput testID={selectTestId} list={mockData} label={defaultLabel} />
    )

    // Select Modal Acilir
    const selectInputValueContainer = getByTestId(selectSelectionTestId)
    waitFor(() => {
      fireEvent.press(selectInputValueContainer)
    })

    const listElement = getByTestId(listTestId)
    mockData.map((item) => {
      expect(listElement).toHaveTextContent(item.label)
    })
  })

  it('default value verildiginde varsayilan deger duzgun sekilde almali', async () => {
    const {
      select: {selectedTestId},
    } = testId
    const onSelectChangeMock = jest.fn()
    const {getByTestId} = render(
      <SelectInput
        list={mockData}
        defaultValue='apple'
        label={defaultLabel}
        value='apple'
        onSelectChange={onSelectChangeMock}
      />
    )

    const selectedItemElement = getByTestId(selectedTestId)

    expect(selectedItemElement.children[0]).toBe('Apple')
  })

  it('secim yapildiginda deger dogru sekilde atanmali', async () => {
    const onSelectChangeMock = jest.fn()
    const {
      select: {selectTestId, selectSelectionTestId, selectedTestId},
    } = testId
    const rendered = render(
      <SelectInput
        testID={selectTestId}
        list={mockData}
        label={defaultLabel}
        onSelectChange={onSelectChangeMock}
      />
    )

    const {getByTestId, getByText} = rendered

    // Select Modal Acilir
    const selectInputValueContainer = getByTestId(selectSelectionTestId)
    await waitFor(() => {
      fireEvent.press(selectInputValueContainer)
    })

    const listElement = getByText('Apple')
    await waitFor(() => {
      fireEvent.press(listElement)
    })

    const selectedElement = getByTestId(selectedTestId)
    expect(selectedElement.children[0]).toBe('Apple')
  })

  it('searchable ozelligi aktif ise search inputu ekranda olmali', () => {
    const onSelectChangeMock = jest.fn()
    const {
      select: {selectTestId, selectSelectionTestId},
      search: {searchInputTestId},
    } = testId
    const {getByTestId, queryByTestId} = render(
      <SelectInput
        testID={selectTestId}
        label={defaultLabel}
        list={mockData}
        onSelectChange={onSelectChangeMock}
        searchable
      />
    )

    // Select Modal Acilir
    const selectInputValueContainer = getByTestId(selectSelectionTestId)
    waitFor(() => {
      fireEvent.press(selectInputValueContainer)
    })

    const searchInputElement = queryByTestId(searchInputTestId)

    expect(searchInputElement).toBeOnTheScreen()
  })

  it('arama islemi aktif degil ise arama inputu ekranda olmamali', () => {
    const onSelectChangeMock = jest.fn()
    const {
      select: {selectTestId, selectSelectionTestId},
      search: {searchInputTestId},
    } = testId
    const {getByTestId, queryByTestId} = render(
      <SelectInput
        testID={selectTestId}
        label={defaultLabel}
        list={mockData}
        onSelectChange={onSelectChangeMock}
      />
    )

    // Select Modal Acilir
    const selectInputValueContainer = getByTestId(selectSelectionTestId)
    waitFor(() => {
      fireEvent.press(selectInputValueContainer)
    })

    const searchInputElement = queryByTestId(searchInputTestId)

    expect(searchInputElement).not.toBeOnTheScreen()
  })

  it('arama islemi aktif ise arama inputu ekranda olmali', () => {
    const onSelectChangeMock = jest.fn()
    const {
      select: {selectTestId, selectSelectionTestId},
      search: {searchInputTestId},
    } = testId
    const rendered = render(
      <SelectInput
        testID={selectTestId}
        label={defaultLabel}
        list={mockData}
        onSelectChange={onSelectChangeMock}
        searchable
      />
    )
    const {getByTestId} = rendered

    // Select Modal Acilir
    const selectInputValueContainer = getByTestId(selectSelectionTestId)
    waitFor(() => {
      fireEvent.press(selectInputValueContainer)
    })

    const searchInputElement = getByTestId(searchInputTestId)

    expect(searchInputElement).toBeOnTheScreen()
  })

  it('arama islemi duzgun filtrelemeli', () => {
    const onSelectChangeMock = jest.fn()
    const {
      select: {selectTestId, selectSelectionTestId, listTestId},
      search: {searchInputTestId},
    } = testId
    const rendered = render(
      <SelectInput
        testID={selectTestId}
        label={defaultLabel}
        list={mockData}
        onSelectChange={onSelectChangeMock}
        searchable
      />
    )
    const {getByTestId} = rendered

    // Select Modal Acilir
    const selectInputValueContainer = getByTestId(selectSelectionTestId)
    waitFor(() => {
      fireEvent.press(selectInputValueContainer)
    })

    // Arama islemi
    const searchInputElement = getByTestId(searchInputTestId)
    waitFor(() => {
      fireEvent.changeText(searchInputElement, 'App')
    })

    const listElement = getByTestId(listTestId)

    //Aranan kelimeye ait veri ekranda olmali
    expect(listElement).toHaveTextContent('Apple')
    //Aramayla alakasiz olan veri ekranda olmamali
    expect(listElement).not.toHaveTextContent('Oppo')
  })

  it('arama islemi sonrasi secim yapildiysa secim duzgun islemeli', () => {
    const onSelectChangeMock = jest.fn()
    const {
      select: {selectTestId, selectSelectionTestId, listTestId, selectedTestId},
      search: {searchInputTestId},
    } = testId
    const rendered = render(
      <SelectInput
        testID={selectTestId}
        label={defaultLabel}
        list={mockData}
        onSelectChange={onSelectChangeMock}
        searchable
      />
    )
    const {getByTestId, getByText} = rendered

    // Select Modal Acilir
    const selectInputValueContainer = getByTestId(selectSelectionTestId)
    waitFor(() => {
      fireEvent.press(selectInputValueContainer)
    })

    // Arama islemi
    const searchInputElement = getByTestId(searchInputTestId)
    waitFor(() => {
      fireEvent.changeText(searchInputElement, 'App')
    })

    const listElement = getByTestId(listTestId)

    // Aranilan Degere Ait Text Ekranda Olmali
    expect(listElement).toHaveTextContent('Apple')

    // Aranilan veriyi secme islemi
    const item = getByText('Apple')
    waitFor(() => {
      fireEvent.press(item)
    })

    const selectedItem = getByTestId(selectedTestId)
    expect(selectedItem.children[0]).toBe('Apple')
  })

  it('arama icin herhangi bir text yazildi ise Clear Iconu ekranda olmali', () => {
    const {
      select: {selectTestId, selectSelectionTestId},
      search: {searchInputTestId, searchClearTestId},
    } = testId
    const rendered = render(
      <SelectInput testID={selectTestId} label={defaultLabel} list={mockData} searchable />
    )
    const {getByTestId} = rendered

    // Select Modal Acilir
    const selectInputValueContainer = getByTestId(selectSelectionTestId)
    waitFor(() => {
      fireEvent.press(selectInputValueContainer)
    })

    // Arama islemi
    const searchInputElement = getByTestId(searchInputTestId)
    waitFor(() => {
      fireEvent.changeText(searchInputElement, 'App')
    })

    // Temizleme Butonu
    let clearInputElement = getByTestId(searchClearTestId)

    expect(clearInputElement).toBeTruthy()
    expect((clearInputElement?.children[0] as any).props.name).toMatch(/REMOVE_CIRCLE/i)
  })

  it('clear iconuna tiklandiginda searchText icerigi temizlenmeli', async () => {
    const {
      select: {selectTestId, selectSelectionTestId},
      search: {searchInputTestId, searchClearTestId},
    } = testId
    const rendered = render(
      <SelectInput testID={selectTestId} label={defaultLabel} list={mockData} searchable />
    )
    const {getByTestId} = rendered

    // Select Modal Acilir
    const selectInputValueContainer = getByTestId(selectSelectionTestId)
    await waitFor(() => {
      fireEvent.press(selectInputValueContainer)
    })

    // Arama islemi
    const searchInputElement = getByTestId(searchInputTestId)
    await waitFor(() => {
      fireEvent.changeText(searchInputElement, 'App')
    })

    // Arama islemini temizler
    await act(() => {
      let clearInputElement = getByTestId(searchClearTestId)
      fireEvent.press(clearInputElement)
    })

    const searchInputElementAfterRender = getByTestId(searchInputTestId)
    expect(searchInputElementAfterRender).toHaveProp('value', '')
  })

  it('clear iconu ile icerik temizlendiginde liste orjinal haline donmeli', () => {
    const {
      select: {selectTestId, selectSelectionTestId, listTestId},
      search: {searchInputTestId, searchClearTestId},
    } = testId
    const rendered = render(
      <SelectInput testID={selectTestId} label={defaultLabel} list={mockData} searchable />
    )
    const {getByTestId} = rendered

    // Select Modal Acilir
    const selectInputValueContainer = getByTestId(selectSelectionTestId)
    waitFor(() => {
      fireEvent.press(selectInputValueContainer)
    })

    // Arama islemi
    const searchInputElement = getByTestId(searchInputTestId)
    waitFor(() => {
      fireEvent.changeText(searchInputElement, 'App')
    })

    // Arama islemini temizler
    let clearInputElement = getByTestId(searchClearTestId)
    waitFor(() => {
      fireEvent.press(clearInputElement)
    })

    // Tum liste ekranda var mi yok mu kiyaslar
    const listElement = getByTestId(listTestId)
    mockData.slice(0, 10).map((item) => {
      expect(listElement).toHaveTextContent(item.label)
    })
  })

  it('secim yapildiginda description parametresi var ise ekranda renderlanmali', async () => {
    const onSelectChangeMock = jest.fn()
    const {
      select: {selectTestId, selectSelectionTestId, selectedDescriptionTestId},
    } = testId
    const rendered = render(
      <SelectInput
        testID={selectTestId}
        list={mockData}
        label={defaultLabel}
        description
        onSelectChange={onSelectChangeMock}
      />
    )

    const {getByTestId, getByText} = rendered

    // Select Modal Acilir
    const selectInputValueContainer = getByTestId(selectSelectionTestId)
    await waitFor(() => {
      fireEvent.press(selectInputValueContainer)
    })

    const listElement = getByText('Apple')
    await waitFor(() => {
      fireEvent.press(listElement)
    })

    const selectedDescElement = getByTestId(selectedDescriptionTestId)
    expect(selectedDescElement.children[0]).toBe('Apple Desc')
  })

  it('description propstan string deger aldiysa ekranda renderlanmali', async () => {
    const {
      select: {selectTestId, selectedDescriptionTestId},
    } = testId
    const mockDesc = 'Test Desc'

    const rendered = render(
      <SelectInput
        testID={selectTestId}
        list={mockData}
        label={defaultLabel}
        description={mockDesc}
      />
    )

    const {getByTestId} = rendered

    const selectedDescElement = getByTestId(selectedDescriptionTestId)
    expect(selectedDescElement.children[0]).toBe(mockDesc)
  })

  it('select input propstan ve list ten es zamanli description almis ise secim yokken props degeri secim var ise data degeri description olarak atanmali', async () => {
    const onSelectChangeMock = jest.fn()
    const {
      select: {selectTestId, selectedTestId, selectSelectionTestId, selectedDescriptionTestId},
    } = testId
    const mockDesc = 'Test Desc'

    const rendered = render(
      <SelectInput
        testID={selectTestId}
        list={mockData}
        label={defaultLabel}
        description={mockDesc}
        showDescription
        onSelectChange={onSelectChangeMock}
      />
    )

    const {getByTestId, getByText} = rendered

    // Mock Description Text Control
    const selectedDescElement = getByTestId(selectedDescriptionTestId)
    expect(selectedDescElement.children[0]).toBe(mockDesc)

    // Select Modal Acilir
    const selectInputValueContainer = getByTestId(selectSelectionTestId)
    await waitFor(() => {
      fireEvent.press(selectInputValueContainer)
    })

    // Bir Deger Secilir
    const listElement = getByText('Apple')
    await waitFor(() => {
      fireEvent.press(listElement)
    })

    const selectedElement = getByTestId(selectedTestId)
    const selectedDescElementRerender = getByTestId(selectedDescriptionTestId)

    expect(selectedElement.children[0]).toBe('Apple')
    expect(selectedDescElementRerender.children[0]).toBe('Apple Desc')
  })
})
