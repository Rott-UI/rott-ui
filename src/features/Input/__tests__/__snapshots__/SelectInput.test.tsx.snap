// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Input -> Custom Input ilk render anında snapshot ile eşleşmeli 1`] = `
<View
  style={
    {
      "alignItems": undefined,
      "backgroundColor": undefined,
      "bottom": undefined,
      "flex": undefined,
      "flexDirection": "column",
      "flexGrow": undefined,
      "gap": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "left": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "maxHeight": undefined,
      "maxWidth": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "overflow": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "paddingVertical": undefined,
      "position": undefined,
      "right": undefined,
      "top": undefined,
      "width": undefined,
      "zIndex": undefined,
    }
  }
  testID="select-test-id"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    alignItemsCenter={true}
    animated={false}
    collapsable={false}
    flex={0}
    focusable={true}
    justifyContentCenter={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    size="full"
    style={
      {
        "alignItems": "center",
        "backgroundColor": undefined,
        "bottom": undefined,
        "flex": 0,
        "flexDirection": "column",
        "flexGrow": undefined,
        "gap": undefined,
        "height": undefined,
        "justifyContent": "center",
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "overflow": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": undefined,
        "right": undefined,
        "top": undefined,
        "width": "100%",
        "zIndex": undefined,
      }
    }
    testID="select-input-selection-test-id"
  >
    <View
      alignItemsCenter={true}
      height={89}
      style={
        {
          "alignItems": "center",
          "backgroundColor": undefined,
          "bottom": undefined,
          "flex": undefined,
          "flexDirection": "row",
          "flexGrow": undefined,
          "gap": undefined,
          "height": 171,
          "justifyContent": undefined,
          "left": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "maxHeight": 171,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "overflow": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "top": undefined,
          "width": undefined,
          "zIndex": undefined,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          {
            "backgroundColor": undefined,
            "bottom": undefined,
            "color": "rgba(161, 173, 175, 1)",
            "flexGrow": undefined,
            "fontFamily": "Markpro-Medium",
            "fontSize": 16,
            "fontWeight": undefined,
            "gap": undefined,
            "height": undefined,
            "left": undefined,
            "letterSpacing": 0.5,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "maxHeight": undefined,
            "maxWidth": undefined,
            "minHeight": undefined,
            "minWidth": undefined,
            "overflow": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "right": undefined,
            "textAlign": "auto",
            "top": undefined,
            "width": "90%",
            "zIndex": undefined,
          }
        }
        testID="select-input-selected-item-test-id"
      >
        Dropdown
      </Text>
      <View
        style={
          {
            "alignItems": undefined,
            "backgroundColor": undefined,
            "bottom": undefined,
            "flex": undefined,
            "flexDirection": "column",
            "flexGrow": undefined,
            "gap": undefined,
            "height": undefined,
            "justifyContent": undefined,
            "left": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "maxHeight": undefined,
            "maxWidth": undefined,
            "minHeight": undefined,
            "minWidth": undefined,
            "overflow": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "right": undefined,
            "top": undefined,
            "width": undefined,
            "zIndex": undefined,
          }
        }
      >
        <SvgMock
          fill="transparent"
          height={48}
          stroke="rgba(161, 173, 175, 1)"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={1}
          width={48}
        />
      </View>
    </View>
  </View>
</View>
`;
